app/controllers/content_items_controller.rb

```diff
@@ -199,4 +200,23 @@ def error_redirect(exception)
     )
     redirect_to destination, status: status_code
   end
+
+  def save_this_page_enabled?
+    ENV["FEATURE_FLAG_SAVE_A_PAGE"] == "enabled"
+  end
+  helper_method :save_this_page_enabled?
+
+  helper_method :logged_in?
+
+  def user_has_saved_page?
+    GdsApi.account_api.get_saved_page(
+      page_path: request.path,
+      govuk_account_session: @account_session_header,
+    ).present?
```

Comment:
You'll want to check the `"saved_page"` field: https://github.com/alphagov/account-api/blob/main/app/controllers/saved_pages_controller.rb#L13

Reply:
Was a bit lazy here, but only becuase it 404s / 401s if something goes wrong.

So a sucessful query here by definition has proved presense?
So we can return true.

But sure will adjust this!